packages/headless/src/features/product-listing/product-listing-analytics.actions.ts

```diff
@@ -0,0 +1,32 @@
+import {Result} from '../..';
+import {ProductListingAnalyticsProvider} from '../../api/analytics/product-listing-analytics';
+import {
+  AnalyticsType,
+  ClickAction,
+  documentIdentifier,
+  makeAnalyticsAction,
+  partialDocumentInformation,
+  ProductListingAction,
+  validateResultPayload,
+} from '../analytics/analytics-utils';
+
+export const logProductListing = (): ProductListingAction =>
+  makeAnalyticsAction(
+    'analytics/productListing/load',
+    AnalyticsType.Search,
+    (client) => client.makeInterfaceLoad(),
```

Comment:
I'm not very familiar with analytics, should I add a `makeListingInterfaceLoad` function to [coveo.analytics.js](https://github.com/coveo/coveo.analytics.js) or is this the generic one ?

Reply:
`makeInterfaceLoad` will log the actionCause `interfaceLoad`, and `makeRecommendationInterfaceLoad` (used by `product-recommendations`) will log the action cause `recommendationInterfaceLoad`.

Ultimately, I guess it's up to the UA team which is better. I don't know if product listing is different enough from search that it needs a different event, or if there's enough users tracking the current event with product listing that changing it would be bad.

Reply:
With the search engine, we provide an `executeFirstSearch` method on the engine. Calling the method dispatches a) a search with b) interfaceLoad as the analytics event.

I think the same event can be reused with the first product listing search. Would name it the same for consistency though,